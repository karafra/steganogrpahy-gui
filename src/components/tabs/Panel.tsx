import React, { ReactNode } from "react";

export interface PanelProps {
  title: string;
  children: ReactNode[] | ReactNode;
  width?: string; 
}

const Panel = (props: PanelProps) => {
  return <div style={{ alignContent: "center", justifyContent: "center" }}>{props.children}</div>;
};

export default Panel;
