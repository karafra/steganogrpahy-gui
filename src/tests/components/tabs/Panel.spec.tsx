import React from 'react';
import renderer from "react-test-renderer";
import Panel from "../../../components/tabs/Panel";

describe('Panel tests', () => {
  
  const testTitle = "Test title"; 
  const testChild = "Test child"
  beforeEach(() => {
    jest.resetAllMocks();
    jest.restoreAllMocks();
  });

  test('should load component', () => {
    // Given
    const component = renderer.create(
    <Panel title={testTitle}>
        {testChild}
    </Panel>  
    );
    // When
    const tree = component.toJSON()
    // Then
    expect(tree).toMatchSnapshot();
  });
});
