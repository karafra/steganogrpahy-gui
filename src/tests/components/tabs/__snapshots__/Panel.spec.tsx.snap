// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Panel tests should load component 1`] = `
<div
  style={
    Object {
      "alignContent": "center",
      "justifyContent": "center",
    }
  }
>
  Test child
</div>
`;
